ssl/t1_lib.c

```diff
@@ -1274,10 +1276,8 @@ unsigned char *ssl_add_clienthello_tlsext(SSL *s, unsigned char *buf,
          * 2 for hostname length
          * + hostname length
          */
-
-        if ((lenmax = limit - ret - 9) < 0
-            || (size_str =
-                strlen(s->tlsext_hostname)) > (unsigned long)lenmax)
+        size_str = strlen(s->tlsext_hostname);
+        if (CHECKLEN(ret, 9 + size_str, limit))
```

Comment:
I think ```CHECKLEN``` is a not-great name, ```TOOBIG``` or similar is better.  See https://www.lysator.liu.se/c/pikestyle.html on function names :)  But that's up to you.